spinalcordtoolbox/scripts/sct_propseg.py

```diff
@@ -497,7 +497,7 @@ def propseg(img_input, options_dict):
         elif str(arguments.init_centerline) == "hough":
             use_optic = False
         else:
-            if rescale_header is not 1:
+            if rescale_header != 1:
```

Comment:
The default value for `rescale` is `1.0`:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/49d7303a09ceef93abd2cdd70d78d9191b454506/spinalcordtoolbox/scripts/sct_propseg.py#L299-L307

So, the changes in this PR result in different behaviour:

```
>>> 1.0 is not 1
True
>>>  1.0 != 1
False
```

However, it should have never been evaluating to True in the first place! So, these checks have been incorrect this whole time. :facepalm: